app/components/admin/custom_fields/hierarchy/change_item_parent_dialog_component.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Admin
+  module CustomFields
+    module Hierarchy
+      class ChangeItemParentDialogComponent < ApplicationComponent
+        include OpTurbo::Streamable
+
+        TEST_SELECTOR = "op-custom-fields--change-item-parent-dialog"
+
+        def initialize(custom_field:, hierarchy_item:)
+          super
+          @custom_field = custom_field
+          @hierarchy_item = hierarchy_item
+        end
+
+        def dialog_id = "custom-fields--change-item-parent-dialog"
+
+        def form_id = "custom-fields--change-item-parent-form"
+
+        def form_arguments
+          {
+            form_id:,
+            url: change_parent_custom_field_item_path(custom_field_id: @custom_field.id, id: @hierarchy_item.id),
+            model: form_model,
+            method: :post
+          }
+        end
+
+        private
+
+        def form_model
+          CustomField::Hierarchy::Forms::NewParentFormModel.new(new_parent: nil)
```

Comment:
I guess it would make sense to initialize this with the current parent by default?

In the current UI this will only show an integer, but in a tree view it would mean that the current parent is already selected.

In both cases it means that cycling through "show the dialog" and "press apply button" is non-destructive.